### PR TITLE
[Integration]: Slack Integration

### PR DESCRIPTION
**Pull request by tcchawla**
## Description

Implemented the Slack API Integration with MindsDB handler

**Fixes** 
#4904

## Type of change

- [X] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?
Implemented `channels` table for the Slack Handler. 
 - Supports Create Database with `Slack` Engine
 - Supports Retrieval of the conversations within the channel
 - Supports message posting in the channel
 - Supports WHERE, ORDER BY, LIMIT clause

### Solution Video
[Demo Video](https://drive.google.com/file/d/1HPosn9qMT_09mBuGizzAio00BQbw__au/view?usp=share_link)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation, or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+                    )
+
+        # Retrieve the conversation history
+        try:
+            result = self.client.conversations_history(channel=params['channel'])
+            conversation_history = result["messages"]
+        except SlackApiError as e:
+            print("Error creating conversation: {}".format(e))
+        
+        print(conversation_history)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:101]: Please remove print

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+            
+        # add absent columns
+        for col in set(columns) & set(result.columns) ^ set(columns):
+            result[col] = None
+
+        # filter by columns
+        columns = [target.parts[-1].lower() for target in query.targets if isinstance(target, ast.Identifier)]
+        result = result[columns]
+
+        print(f"{len(result['messages'])} Messages found in Channel {params['channel']}")
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:131]: Same here

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    
+    def connect(self):
+        """
+        Authenticate with the Slack API using the token stored in the `token` attribute.
+        """
+        if self.is_connected is True:
+            return self.api
+
+        self.api = self.create_connection()
+        print("Connected to Slack API")
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:310]: Same here

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+        response = StatusResponse(False)
+
+        try:
+            api = self.connect()
+
+            # Call API method to check the connection
+            api.auth_test()
+            
+            response.success = True
+            print("Connected to Slack API.")
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:326]: Remove this

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+        except SlackApiError as e:
+            response.error_message = f'Error connecting to Slack Api: {e.response["error"]}. Check token.'
+            print(f"Error connecting to Slack API: {e}")
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:329]: Also here

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+            channel_name
+
+        Returns:
+            conversation_history
+        """
+
+        # Get the channels list and ids
+        channels = self.client.conversations_list(types="public_channel,private_channel")['channels']
+        channel_ids = {c['name']: c['id'] for c in channels}
+        print(channel_ids)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:48]: Please remove print

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+                    raise ValueError(
+                        f"Order by unknown column {an_order.field.parts[1]}"
+                    )
+
+        # Retrieve the conversation history
+        try:
+            result = self.client.conversations_history(channel=params['channel'])
+            conversation_history = result["messages"]
+        except SlackApiError as e:
+            print("Error creating conversation: {}".format(e))
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:99]: Let's change to log

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+            # post message to Slack channel
+            try:
+                response = self.client.chat_postMessage(
+                    channel=params['channel'],
+                    text=params['message']
+                )
+            except SlackApiError as e:
+                raise Exception(f"Error posting message to Slack channel '{params['channel']}': {e.response['error']}")
+            
+            print(response)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:200]: Remove this

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 9]
```diff
@@ -0,0 +1,396 @@
...
+        try:
+            response = self.client.chat_update(
+                channel=params['channel'],
+                ts=params['ts'],
+                text=params['message']
+            )
+        except SlackApiError as e:
+            raise Exception(f"Error updating message in Slack channel '{params['channel']}' with timestamp '{params['ts']}': {e.response['error']}")
+        
+        print(response)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:233]: Remove this

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py [hunk 10]
```diff
@@ -0,0 +1,396 @@
...
+        try:
+            response = self.client.chat_delete(
+                channel=params['channel'],
+                ts=params['ts']
+            )
+            
+        except SlackApiError as e:
+            raise Exception(f"Error deleting message from Slack channel '{params['channel']}' with timestamp '{params['ts']}': {e.response['error']}")
+
+        print(response)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/slack_handler/slack_handler.py:263]: Remove this

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 7 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py:99]: Or raise Error

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 1 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 2 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 3 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 4 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 6 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 7 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here, and changed it to log.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 9 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

File: mindsdb/integrations/handlers/slack_handler/slack_handler.py (hunk 10 above)
Comment (tcchawla) [reply mindsdb/integrations/handlers/slack_handler/slack_handler.py]: Removed print here.

**Review by ZoranPandovski — APPROVED** [review]

**Comment by ZoranPandovski** [comment]
Thanks. Tests are failing due to dependency issue not related with this PR, solved with https://github.com/mindsdb/mindsdb/pull/6031

**Comment by tcchawla** [comment]
Thanks ❤️ @ZoranPandovski  @StpMax

**Pull request merged**